src/Tribe/Repositories/Event.php

```diff
@@ -1463,26 +1460,49 @@ public function handle_order_by( $order_by ) {
 
 		$timestamp_key = 'TIMESTAMP(mt1.meta_value)';
 
-		$by_event_start_date     = isset( $check_orderby['event_date'] )
-		                           || in_array( 'event_date', $check_orderby, true );
-		$by_event_start_date_utc = isset( $check_orderby['event_date_utc'] )
-		                           || in_array( 'event_date_utc', $check_orderby, true );
-		$by_organizer            = isset( $check_orderby['organizer'] )
-		                           || in_array( 'organizer', $check_orderby, true );
-		$by_venue                = isset( $check_orderby['venue'] ) || in_array( 'venue', $check_orderby, true );
-		$by_timestamp_key        = isset( $check_orderby[ $timestamp_key ] )
-		                           || in_array( $timestamp_key, $check_orderby, true );
-
-		if ( $by_event_start_date || $by_event_start_date_utc ) {
-			$this->order_by_date( $by_event_start_date_utc );
-		} elseif ( $by_organizer ) {
-			$this->order_by_organizer();
-		} elseif ( $by_venue ) {
-			$this->order_by_venue();
-		} elseif ( $by_timestamp_key ) {
-			$this->filter_query->orderby( $timestamp_key );
-		} else {
-			$this->query_args['orderby'] = $order_by;
+		$after    = false;
+		$loop = 0;
+
+		foreach ( $check_orderby as $key => $value ) {
+			$loop++;
+			$order_by      = is_numeric( $key ) ? $value : $key;
+			$order         = is_numeric( $key ) ? 'ASC' : $value;
+			$default_order = Arr::get_in_any( [ $this->query_args, $this->default_args ], 'order', 'ASC' );
+
+			switch ( $order_by ) {
+				case 'event_date':
+					$this->order_by_date( false, $after );
+					break;
+				case 'event_date_utc':
+					$this->order_by_date( true, $after );
+					break;
+				case 'organizer':
+					$this->order_by_organizer( $after );
+					break;
+				case 'venue':
+					$this->order_by_venue( $after );
+					break;
+				case $timestamp_key:
+					$this->filter_query->orderby( [ $timestamp_key => $default_order ], null, null, $after );
+					break;
+				default:
+					$after = $after || $loop === 1;
```

Comment:
:no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: 73015339f395aa39a4ce7ee1e33c2010</sub>